test/CMakeLists.txt

```diff
@@ -81,7 +81,7 @@ if(MSVC)
     endif()
 else()
     # Non-MSVC specific configuration (original content)
-    set(CMAKE_CXX_STANDARD 17)
+    set(CMAKE_CXX_STANDARD 23)
```

Comment:
Support for compilation under MSVC is on a "best effort" basis, we try, but we can't guarantee we will succeed at all the things.  It is acceptable to bump to 23, but please explain why you found the need

Reply:
This is for the non-MSVC path. This is to enable support for std::array in a constexpr context.